plugins/woocommerce-admin/client/marketplace/components/search-results/search-results.tsx

```diff
@@ -22,50 +30,112 @@ export default function SearchResults( props: SearchResultProps ): JSX.Element {
 	const themeList = props.products.filter(
 		( product ) => product.type === ProductType.theme
 	);
+	const marketplaceContextValue = useContext( MarketplaceContext );
+	const { isLoading } = marketplaceContextValue;
 
 	const query = useQuery();
 	const showCategorySelector = query.section ? true : false;
 	const searchTerm = query.term ? query.term : '';
 
-	const extensionComponent = (
-		<Products
-			products={ extensionList }
-			type={ ProductType.extension }
-			categorySelector={ showCategorySelector }
-			searchTerm={ searchTerm }
-		/>
-	);
+	type Overrides = {
+		categorySelector?: boolean;
+		showAllButton?: boolean;
+		perPage?: number;
+	};
 
-	const themeComponent = (
-		<Products
-			products={ themeList }
-			type={ ProductType.theme }
-			categorySelector={ showCategorySelector }
-			searchTerm={ searchTerm }
-		/>
-	);
+	function productsComponent(
+		products: Product[],
+		type: ProductType,
+		overrides: Overrides = {}
+	) {
+		return (
+			<Products
+				products={ products }
+				type={ type }
+				categorySelector={
+					overrides.categorySelector ?? showCategorySelector
+				}
+				searchTerm={ searchTerm }
+				showAllButton={ overrides.showAllButton ?? true }
+				perPage={ overrides.perPage ?? MARKETPLACE_ITEMS_PER_PAGE }
+			/>
+		);
+	}
+
+	function extensionsComponent( overrides: Overrides = {} ) {
+		return productsComponent(
+			extensionList,
+			ProductType.extension,
+			overrides
+		);
+	}
+
+	function themesComponent( overrides: Overrides = {} ) {
+		return productsComponent( themeList, ProductType.theme, overrides );
+	}
 
 	const content = () => {
-		if ( query?.section === SearchResultType.theme ) {
-			return themeComponent;
-		}
 		if ( query?.section === SearchResultType.extension ) {
-			return extensionComponent;
+			return extensionsComponent( { showAllButton: false } );
 		}
 
-		if ( extensionList.length === 0 && themeList.length > 0 ) {
+		if ( query?.section === SearchResultType.theme ) {
+			return themesComponent( { showAllButton: false } );
+		}
+
+		// Components can handle their isLoading state. So we can put them both on the page.
```

Comment:
These comments are a really great example of useful context. I could see it being easy to miss that the loading state is here, but that the components handle how they appear when loading separately.

Reply:
Thanks! This is one of the parts of the code I'm happy with, for the reason you have pointed out. It's very easy to miss. Comments for the rescue until we fix this.